sdk/identity/azure-identity/src/client_secret_credential.cpp

```diff
@@ -80,27 +80,37 @@ Azure::Core::Credentials::AccessToken ClientSecretCredential::GetToken(
     }
   }
 
-  return TokenCache::GetToken(m_tenantId, m_clientId, m_authorityHost, scopesStr, [&]() {
-    return m_tokenCredentialImpl->GetToken(context, [&]() {
-      using Azure::Core::Http::HttpMethod;
-
-      std::ostringstream body;
-      body << m_requestBody;
-
-      if (!scopesStr.empty())
-      {
-        body << "&scope=" << scopesStr;
-      }
-
-      auto request = std::make_unique<TokenCredentialImpl::TokenRequest>(
-          HttpMethod::Post, m_requestUrl, body.str());
-
-      if (m_isAdfs)
-      {
-        request->HttpRequest.SetHeader("Host", m_requestUrl.GetHost());
-      }
-
-      return request;
-    });
-  });
+  // TokenCache::GetToken() and m_tokenCredentialImpl->GetToken() can only use the lambda argument
+  // when they are being executed. They are not supposed to keep a reference to lambda argument to
+  // call it later. Therefore, any capture made here will outlive the possible time frame when the
+  // lambda might get called.
+  return TokenCache::GetToken(
+      m_tenantId,
+      m_clientId,
+      m_authorityHost,
+      scopesStr,
+      tokenRequestContext.MinimumExpiration,
```

Comment:
Would it make more sense to pass the `tokenRequestContext` here to future-proof other uses of context values?

Reply:
We could, but I rather wouldn't for now - `tokenRequestContext` contains scopes vector. But we get scopes already formatted as string, ready to go to a network request, or to a map key. Although we'd get it by ref, it may get a bit confusing, plus the test code is not going to look as pretty. And it is a code in `_detail` namespace, not exposed to anyone, being used internally only within `Identity` lib, so there's no immediate need to future-proof at this point. If we make it `_internal`, or  - should there come more parameters that need to go into that function - I think that whoever makes that change would go through that thinking process naturally and will likely replace `minimumExpiration` with `TokenRequestContext const& tokenRequestContext`. I'm 51/49 on this, if you feel more strongly towards future-proofing, I can do that.